youtube_dl/options.py

```diff
@@ -350,6 +350,10 @@ def _scrub_eq(o):
         '--video-password',
         dest='videopassword', metavar='PASSWORD',
         help='Video password (vimeo, smotri, youku)')
+    authentication.add_option(
+        '--auth-provider',
+        dest='auth_provider', metavar='AUTH_PROVIDER',
+        help='Authentication provider (nhl, rogers, ...)')
```

Comment:
I was suggesting that it be a generic option, perhaps it would be nice to deprecate the similar `--ap-mso` option in favor of this more descriptive option.

Happy to make this only for the nhl plugin if that's more appropriate.